framelesswindow/framelesswindow.cpp

```diff
@@ -43,17 +48,50 @@ FramelessWindow::FramelessWindow(QWidget *parent): QWidget(parent)
   windowFrame->setGraphicsEffect(windowShadow);
 
   QObject::connect(qApp, &QGuiApplication::applicationStateChanged, this, &FramelessWindow::on_applicationStateChanged);
+  setMouseTracking(true);
+  
+  //important to watch mouse move from all child widgets
+  QApplication::instance()->installEventFilter(this);
+}
+
+void FramelessWindow::on_restoreButton_clicked()
+{
+	restoreButton->setVisible(false);

```

Comment:
Just one minor hint here:
can you please also replace all "tabs" by two blanks, because it is the code formatting for this sources.